### DISCUSSION TITLE
How to I add a custom snippet?

### DISCUSSION THREAD

**Discussion by 0xF5T9** [LazyVim/LazyVim#3294] [Q&A] [answered]
I can't figure it out how to add a custom snippet.
I checked the documentation: https://www.lazyvim.org/plugins/coding#nvim-snippets
 and found "[nvim-snippets](https://github.com/garymjr/nvim-snippets)", which seem to be what I was looking for.
 
https://github.com/garymjr/nvim-snippets - This mentioned the default "search_paths" is "{vim.fn.stdpath('config') .. '/snippets'}" (nvim/snippets)

So I went to the nvim folder and create "snippets" folder and add a snippet file (i just named it "test"):
```
{
  "Say hello to the world": {
    "prefix": ["hw", "hello"],
    "body": "Hello, ${1:world}!$0"
  }
}
```
But it's not working, no snippet is shown when I type the prefix. I'm new to Vim, what am I doing wrong?

**Comment by dpetka2001** [comment, marked as answer]
You also need to create a `package.json` file which defines the snippets for the languages you have (that's just how VsCode style snippets work). I have created a minimal structure for playing around with native snippets in my [dotfiles here](https://github.com/dpetka2001/dotfiles/tree/main/dot_config/nvim/misc/snippets). Take a look to see how you can configure it in your own configuration.

**Reply by 0xF5T9** [reply]
Thanks. I just check this repository of your and it worked:
https://github.com/dpetka2001/dotfiles/tree/main/dot_config/nvim/misc/snippets
